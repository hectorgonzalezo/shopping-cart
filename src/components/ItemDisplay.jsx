import React from 'react';
import { useParams } from 'react-router-dom';
import PropTypes from 'prop-types';
import uniqid from 'uniqid';
import products from '../assets/products.json';

function ItemDisplay({ opaque }) {
  const { id } = useParams();
  // get product corresponding to id from json file
  const selectedProduct = Object.values(products).filter((product) => product.id === id);
  // get relevant values
  const { img, name, speed, signal, description } = selectedProduct[0];
  // Split description into paragraphs
  const descriptionParagraphs = description.split('\n');
  return (
    <article className={`item-display ${opaque ? 'opaque' : ''}`}>
      <img src={require(`../assets/products/${img}`)} />
      <h1>{name}</h1>
      {speed !== undefined ? <h2>{speed}</h2> : null}
      {signal !== undefined ? <h2>{signal}</h2> : null}
      {descriptionParagraphs.map((paragraph) => (
        <p key={uniqid()} className="paragraph">{paragraph}</p>
      ))}
    </article>
  );
}

ItemDisplay.defaultProps = {
  opaque: false,
};

ItemDisplay.propTypes = {
  opaque: PropTypes.bool,
};

export default ItemDisplay;