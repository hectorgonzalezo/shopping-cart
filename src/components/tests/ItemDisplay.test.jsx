import React from 'react';
import { getByTestId, render, screen } from '@testing-library/react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import '@testing-library/jest-dom';
import ItemDisplay from '../ItemDisplay';

// mock function used to mimick id parameter
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
  useParams: () => ({
    id: '1',
  }),
  useRouteMatch: () => ({ url: '/company/company-id1/team/team-id1' }),
}));

describe('Item Display component', () => {
  it('Renders correctly', () => {
    render(<ItemDisplay />);

    const itemDisplayed = screen.getByRole('article');

    expect(itemDisplayed).toBeInTheDocument();
    expect(screen.container).toMatchSnapshot();
  });
});