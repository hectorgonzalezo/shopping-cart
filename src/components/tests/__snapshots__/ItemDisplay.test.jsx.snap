// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Display component Renders correctly 1`] = `undefined`;
